cdStages_test.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/devtron-labs/ci-runner/helper"
+	test_data "github.com/devtron-labs/ci-runner/test-data"
+	"github.com/devtron-labs/ci-runner/util"
+	"os"
+	"testing"
+)
+
+func TestHandleCDEvent(t *testing.T) {
+	t.Run("StageYamlNoWithNoError", func(t *testing.T) {
+
+		// Prepare test data
+		ciCdRequest := &helper.CiCdTriggerEvent{}
+		json.Unmarshal([]byte(test_data.CdTriggerEventPayloadWithTaskYaml), ciCdRequest)
+
+		exitCode := 0
+
+		// Call the function
+		HandleCDEvent(ciCdRequest, &exitCode)
+
+		// Assert the expected results
+		if exitCode != 0 {
```

Comment:
How are we doing the validations ??

Reply:
added cases a discussed

--- PASS: TestHandleCDEvent (95.69s)
    --- PASS: TestHandleCDEvent/StageYamlNoWithNoError (18.40s)
    --- PASS: TestHandleCDEvent/StageYamlWithError (10.72s)
    --- PASS: TestHandleCDEvent/StageYamlWithNoArtifact (16.79s)
    --- PASS: TestHandleCDEvent/StepsStageWithNoError (0.60s)
    --- PASS: TestHandleCDEvent/StepsStageVarOutputCheckPass (17.52s)
    --- PASS: TestHandleCDEvent/StepsStageVarOutputCheckFail (11.03s)
    --- PASS: TestHandleCDEvent/StepsStageOutputWithError (17.60s)
    --- PASS: TestHandleCDEvent/StepsStageWithError (1.44s)
    --- PASS: TestHandleCDEvent/StepsStageWithSuccessTriggerCriteria (0.39s)
    --- PASS: TestHandleCDEvent/StepsStagePlugin (1.20s)
=== RUN   TestRunScripts
--- SKIP: TestRunScripts (0.00s)
=== RUN   Test_buildDockerRunCommand
=== RUN   Test_buildDockerRunCommand/all_single
=== RUN   Test_buildDockerRunCommand/all_multi
--- PASS: Test_buildDockerRunCommand (0.00s)
    --- PASS: Test_buildDockerRunCommand/all_single (0.00s)
    --- PASS: Test_buildDockerRunCommand/all_multi (0.00s)
=== RUN   Test_buildDockerEntryScript
=== RUN   Test_buildDockerEntryScript/hello
=== RUN   Test_buildDockerEntryScript/ls_dir
=== RUN   Test_buildDockerEntryScript/ls_dir_with_out
=== RUN   Test_buildDockerEntryScript/ls_dir_with_out_multi
coverage: set 0
--- PASS: Test_buildDockerEntryScript (0.00s)
    --- PASS: Test_buildDockerEntryScript/hello (0.00s)
    --- PASS: Test_buildDockerEntryScript/ls_dir (0.00s)
    --- PASS: Test_buildDockerEntryScript/ls_dir_with_out (0.00s)
    --- PASS: Test_buildDockerEntryScript/ls_dir_with_out_multi (0.00s)
=== RUN   TestRunScriptsInDocker
--- SKIP: TestRunScriptsInDocker (0.00s)
=== RUN   Test_writeToEnvFile
=== RUN   Test_writeToEnvFile/empty_env_map
=== RUN   Test_writeToEnvFile/single_env_var
=== RUN   Test_writeToEnvFile/multiple_env_vars
=== RUN   Test_writeToEnvFile/error_creating_file
2023/07/19 13:04:34 DEVTRON error while creating env file  open /dev/null/abcd: not a directory
--- PASS: Test_writeToEnvFile (0.02s)
    --- PASS: Test_writeToEnvFile/empty_env_map (0.01s)
    --- PASS: Test_writeToEnvFile/single_env_var (0.01s)
    --- PASS: Test_writeToEnvFile/multiple_env_vars (0.01s)
    --- PASS: Test_writeToEnvFile/error_creating_file (0.00s)
PASS
	github.com/devtron-labs/ci-runner	coverage: 46.1% of statements
ok  	github.com/devtron-labs/ci-runner	95.768s	coverage: set 0
=== RUN   TestCommandExecutor
=== RUN   TestCommandExecutor/execute_cmd
exit status 2

--- PASS: TestCommandExecutor (0.04s)
    --- PASS: TestCommandExecutor/execute_cmd (0.04s)
=== RUN   Test_evaluateExpression
=== RUN   Test_evaluateExpression/Eval_false
=== RUN   Test_evaluateExpression/eval_true
=== RUN   Test_evaluateExpression/Eval_true_date
=== RUN   Test_evaluateExpression/Eval_false_date
--- PASS: Test_evaluateExpression (0.00s)
    --- PASS: Test_evaluateExpression/Eval_false (0.00s)
    --- PASS: Test_evaluateExpression/eval_true (0.00s)
    --- PASS: Test_evaluateExpression/Eval_true_date (0.00s)
    --- PASS: Test_evaluateExpression/Eval_false_date (0.00s)
PASS
	github.com/devtron-labs/ci-runner/helper	coverage: 2.3% of statements
ok  	github.com/devtron-labs/ci-runner/helper	0.061s	coverage: 2.3% of statements